src/full/Agda/TypeChecking/Rules/LHS.hs

```diff
@@ -1874,11 +1852,8 @@ disambiguateConstructor ambC@(AmbQ cs) d pars = do
         -- meaning that only the parameters may differ,
         -- then throw more specific error.
         (_    , [_]) -> typeError $ CantResolveOverloadedConstructorsTargetingSameDatatype d cs
-        (_    , disambs@(((c,_,_):|_):_)) -> typeError . GenericDocError =<< vcat
-          [ "Ambiguous constructor " <> pretty (qnameName c) <> "."
-          , "It could refer to any of"
-          , nest 2 $ vcat $ map (prettyDisambCons . conName . snd3) $ List1.concat disambs
-          ]
+        (_    , disambs@(((c,_,_):|_):_)) -> typeError . AmbiguousConstructor c =<<
+          (mapM (prettyDisambCons . conName . snd3) $ List1.concat disambs)
```

Comment:
Can the `AmbiguousConstructor` just get the `conName`s and the printing to `Doc` happens when printing the error?